docs/_snippets/recipes_flutter_navigation/navigation2/main.dart.md

```diff
@@ -0,0 +1,99 @@
+```dart
+import 'package:flow_builder/flow_builder.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'bloc/book_bloc.dart';
+
+void main() {
+  runApp(
+    BlocProvider(
+      create: (context) => BookBloc(),
+      child: BooksApp(),
+    ),
+  );
+}
+
+List<Page> onGeneratePages(BookState state, List<Page> pages) {
+  final selectedBook = state.selectedBook;
+  return [
+    BooksListPage(books: state.books).page(),
+    if (selectedBook != null) BookDetailsPage(book: selectedBook).page()
+  ];
+}
+
+class BooksApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: 'Books App',
+      home: FlowBuilder(
+        state: context.watch<BookBloc>().state,
+        onGeneratePages: onGeneratePages,
+      ),
+    );
+  }
+}
+
+class BooksListPage extends StatelessWidget {
+  Page page() => MaterialPage<void>(
+          child: BooksListPage(
+        books: books,
+      ));
+
+  final List<Book> books;
+  const BooksListPage({
+    Key? key,
+    required this.books,
+  }) : super(key: key);
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(),
+      body: ListView(
+        children: [
+          for (final book in books)
+            ListTile(
+              title: Text(book.title),
+              subtitle: Text(book.author),
+              onTap: () => context.read<BookBloc>().add(
+                    BookSelected(book: book),
+                  ),
+            )
+        ],
+      ),
+    );
+  }
+}
+
+class BookDetailsPage extends StatelessWidget {
+  Page page() => MaterialPage<void>(
+          child: BookDetailsPage(
+        book: book,
+      ));
+
+  final Book book;
+
+  const BookDetailsPage({
```

Comment:
can we sort constructors first?

Reply:
What do you mean?

Reply:
@liPatrick to put constructors at the top, before every property of the class 👍